### PR TITLE
udev/net: support to set MDI-X mode

### LINKED ISSUE #22386: udev: Support for setting MDI-X state in .link files
Low priority (very rarely needed).
If it is possible I would like to be able to set mdi-x state in similar way as speed/duplex can be set. Classic nework scripts (like in Fedora) allow to specify ethtool options in configuration file (ifcfg-xxx) in convenient way (ETHTOOL_OPTS="mdix on").

**Describe the solution you'd like**
Something like MdiX=auto|off|on in [Link] section would be great.

**Describe alternatives you've considered**
Alternatives considered:
- udev rule calling ethtool
- custom .service started after network-online.target calling ethtool
- calling ethtool in rc.local or other startup script

**The systemd version you checked that didn't have the feature you are asking for**
Feature does not exist in systemd-networkd-249.6-2.fc35.x86_64

### PR DESCRIPTION
**Pull request by yuwata**
Closes #22386.

### CODE REVIEW

File: man/systemd.link.xml [hunk 1]
```diff
@@ -946,6 +946,16 @@
         </listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>MDI-X=</varname></term>
+        <listitem>
+          <para>Specifies MDI-X mode for the port. Takes a boolean value or special value
```
Comment (poettering) [inline man/systemd.link.xml]: some brief explanation what MDI-X actually is would be great. After reading this man page I am now more confused than before.

File: man/systemd.link.xml [hunk 2]
```diff
@@ -946,6 +946,16 @@
...
+          <literal>auto</literal>. When true, the MDI-X (crossover) mode will be used. When false, the
+          MDI (straight through) mode will be used. When <literal>auto</literal>, the MDI status is
+          automatically detected. Defaults to unset, and the kernel's default will be used.</para>
+        </listitem>
+      </varlistentry>
```
Comment (poettering) [inline man/systemd.link.xml:962]: so after reading the wikipedia page it appears to me we should normalize this option, i.e.

call it MDI= and then take "mdi" and "mdix" as parameters. i.e. for the same reason the wikipedia page is called "MDI" and not "MDI-X" we should not name our option "MDI-X" either, but just "MDI". 

(might make sense to make this parser a bit flexible and accept the syntaxes "mdi" and "straight" as equivalent, as well as "mdi-x", "midx" and "crossover" as equivalent.)

**Comment by poettering** [comment]
(codewise looks fine too me)

**Comment by GitArUs** [comment]
I have tested copr build, works as expected. Thanks :)

**Comment by yuwata** [comment]
@poettering Thank you for the review. The name of the setting is changed MDI-X= -> MDI=, and takes words like "mdi", "mdi-x", "auto". Also, I copied brief explanation what is MDI from Wikipedia. PTAL.

@GitArUs Thank you for testing the PR so quickly!

**Comment by GitArUs** [comment]
Tested again, works very well.

File: man/systemd.link.xml [hunk 3]
```diff
@@ -946,6 +946,21 @@
...
 
+      <varlistentry>
+        <term><varname>MDI=</varname></term>
+        <listitem>
+          <para>Specifies the medium dependent interface (MDI) mode for the interface. A MDI describes
+          the interface from a physical layer implementation to the physical medium used to carry the
+          transmission. Takes one of the following words: <literal>mdi</literal>,
+          <literal>straight</literal>, <literal>mdi-x</literal>, <literal>mdix</literal>,
+          <literal>crossover</literal>, and <literal>auto</literal>. When <literal>mdi</literal> or
+          <literal>straight</literal>, the MDI straight through mode will be used. When
```
Comment (poettering) [inline man/systemd.link.xml:957]: maybe write this as:

```
…<literal>straight</literal> (or equivalently: <literal>mdi</literal>), <literal>crossover</literal> (or equivalently: <literal>mdi-x</literal>, <literal>mdix</literal>), or <literal>auto</literal>.…
```

i.e. emphasize that these aren't actually six different options, but just three, as three of them are just aliases for two others.

**Comment by poettering** [comment]
lgtm. tiny suggestion for the man page, looks great otherwise.

**Comment by yuwata** [comment]
@poettering Thank you for the review. The man page is updated according to the above comment. Upgrading the green label.

**Pull request merged**